lib/common/utils/document.js

```diff
@@ -118,22 +121,149 @@ define(['jquery', 'async', 'underscore'], function ($, async, _) {
             return this;
         },
 
+        addPageTag: function (ctx, isServer, name, attributes, content) {
+
+            var pageTag = this._createPageTag(name, attributes, content);
+
+            if (isServer) {
```

Comment:
The client uses the rootCtx tags to remove existing page tags. The client moves pageTags from the controller ctx to the root ctx after rendering the tags (see line 170). So the server adds to the rootCtx to ensure the client can remove any tags rendered by the server.

The server could add tags to the controller ctx, but then you'd need to add logic on the server to move tags to the rootCtx after rendering the page, and this seemed more complicated.